x-pack/plugins/maps/server/routes.js

```diff
@@ -540,7 +540,7 @@ export function initRoutes(router, licenseUid, mapConfig, kbnVersion, logger) {
       }
 
       try {
-        const resp = await context.core.elasticsearch.legacy.client.callAsCurrentUser(
+        const resp = await context.core.elasticsearch.legacy.client.callAsInternalUser(
```

Comment:
@nreese thanks for the ping!

We unfortunately can't use `callAsInternalUser` as-is for this route. Doing so would allow any user, regardless of their privileges, to view settings for all indices. By extension, it would also allow them to enumerate available indices, and all fields within those indices, that they otherwise wouldn't be aware of due to their privileges.

If we _really_ need to support users without `view_index_metadata` privileges, then I think we'll need to do this in two steps:

1) Use `callAsCurrentUser` to retrieve the list of indices the user is authorized against which match the provided pattern. Alerting has an example of how to do this here: https://github.com/elastic/kibana/blob/fe1c508d8d42f750f830fc2e3caec5aaa4f66c0d/x-pack/plugins/alerting_builtins/server/alert_types/index_threshold/routes/indices.ts#L86

Once you extract the set of indices from that call, _then_ you can `callAsInternalUser` to get the settings just for the indices returned in step 1 (don't use the provided pattern from the API call).

Then, the result of this second call should be filtered to only return the information you absolutely need. For example, we shouldn't return the full set of fields, since the end-user might not be authorized to see those fields ([Field-level security](https://www.elastic.co/guide/en/elasticsearch/reference/current/field-level-security.html) might be in effect for this user). It looks like you're already doing this step though via your call to `getIndexPatternSettings` 👍